integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/OpenLineageRunEventBuilder.java

```diff
@@ -281,15 +297,49 @@ RunEvent buildRun(
     return buildRun(
         applicationParentRunFacet,
         runEventBuilder,
+        openLineageContext.getRunUuid(),
         jobBuilder,
         jobFacetsBuilder,
         event,
         Optional.ofNullable(jobMap.get(event.jobId())));
   }
 
+  RunEvent buildRun(
+      ParentRunFacet parentRunFacet,
+      RunEventBuilder runEventBuilder,
+      JobBuilder jobBuilder,
+      OpenLineage.JobFacetsBuilder jobFacetsBuilder,
+      SparkListenerApplicationStart event) {
+    return buildRun(
+        parentRunFacet,
+        runEventBuilder,
+        openLineageContext.getApplicationUuid(),
+        jobBuilder,
+        jobFacetsBuilder,
+        event,
+        Optional.empty());
+  }
+
+  RunEvent buildRun(
+      ParentRunFacet applicationParentRunFacet,
+      RunEventBuilder runEventBuilder,
+      JobBuilder jobBuilder,
+      OpenLineage.JobFacetsBuilder jobFacetsBuilder,
+      SparkListenerApplicationEnd event) {
+    return buildRun(
+        applicationParentRunFacet,
+        runEventBuilder,
+        openLineageContext.getApplicationUuid(),
+        jobBuilder,
+        jobFacetsBuilder,
+        event,
+        Optional.empty());
+  }
+
   private RunEvent buildRun(
       ParentRunFacet applicationParentRunFacet,
       RunEventBuilder runEventBuilder,
+      UUID runId,
```

Comment:
I hope one day we'll come up with some nested builder for OL events which will allow replacing all these arguments with a single one.